lib/puppet/application_support.rb

```diff
@@ -19,7 +19,7 @@ module ApplicationSupport
     # @return [void]
     # @api private
     def self.push_application_context(run_mode, environment_mode = :local)
-      Puppet.push_context(Puppet.base_context(Puppet.settings), "Update for application settings (#{run_mode})")
+      Puppet.push_context_global(Puppet.base_context(Puppet.settings), "Update for application settings (#{run_mode})")
```

Comment:
So, will anything that calls `push_application_context` be unsafe at this? Or is this just to test to see if this will fix the `environmentpath` issues?

Reply:
Yeah I guess